src/main/java/seedu/address/model/guest/Guest.java

```diff
@@ -113,4 +113,17 @@ public String toString() {
         return builder.toString();
     }
 
+    @Override
```

Comment:
Which part of Ab requires comparing `Guest`s?

Reply:
`Booking`, which contains `Guest`.
`Bookings` maintains a `SortedSet<Booking>`. We want to compare all fields of `Booking` because the `Set` detects duplication using `compareTo`. In order for us to detect overlapping bookings, we need to be able to add bookings with same booking period but different guest into the sorted set. If this confuses you, consider this: if we not compare the guest, then adding two bookings with same booking period but different guest will simply cause the second one to be considered as duplicate and ignored by the SortedSet, without any indication of duplication. We cannot use `contains` to detect duplication before we add, because `contains` is based on `equals`, and a `Booking` is equal to another only if all fields are equals.